selfdrive/manager.py

```diff
@@ -289,6 +289,15 @@ def prepare_managed_process(p):
       subprocess.check_call(["make", "clean"], cwd=os.path.join(BASEDIR, proc[0]))
       subprocess.check_call(["make", "-j4"], cwd=os.path.join(BASEDIR, proc[0]))
 
+
+def join_process(process, timeout):
+  # Process().join(timeout) will hang due to a python 3 bug: https://bugs.python.org/issue28382
+  # We have to poll the exitcode instead
+  t = time.time()
+  while time.time() - t < timeout and process.exitcode is None:
+    time.sleep(0.001)
```

Comment:
Should it actually join the process after this? Otherwise it's an orphan PID or something.

Reply:
After calling this function manager calls `os.kill` on the process, so it won't be orphaned. Or did you want to move that into the helper function?